### PR TITLE
drivers: sensor: bma280: Update driver to use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by benjaminbjornsson**

### CODE REVIEW

File: drivers/sensor/bma280/bma280.h [hunk 1]
```diff
@@ -141,6 +140,7 @@ struct bma280_data {
 
 struct bma280_config {
 	const struct i2c_dt_spec i2c;
+	const struct gpio_dt_spec int1_gpio;
```
Comment (galak) [inline drivers/sensor/bma280/bma280.h:143]: Add a #ifdef CONFIG_BMA280_TRIGGER around `gpio_dt_spec`

File: drivers/sensor/bma280/bma280.c [hunk 2]
```diff
@@ -164,6 +164,7 @@ static struct bma280_data bma280_inst_data;
 
 static struct bma280_config bma280_inst_config = {
 	.i2c = I2C_DT_SPEC_INST_GET(0),
+	.int1_gpio = GPIO_DT_SPEC_INST_GET(0, int1_gpios),
```
Comment (galak) [inline drivers/sensor/bma280/bma280.c:167]: add an #ifdef around `.int1_gpio`

Comment (gmarull) [reply drivers/sensor/bma280/bma280.c:168]: note: if we're planning to make drivers multi-instance later (we should), this pattern is better since we can't have #ifdefs inside macros:

```c
static struct bma280_config bma280_inst_config = {
	.i2c = I2C_DT_SPEC_INST_GET(0),
	IF_ENABLED(CONFIG_BMA280_TRIGGER,
		   (.int1_gpio = GPIO_DT_SPEC_INST_GET(0, int1_gpios),))
};
```

File: drivers/sensor/bma280/bma280_trigger.c [hunk 3]
```diff
@@ -20,13 +20,13 @@ LOG_MODULE_DECLARE(BMA280, CONFIG_SENSOR_LOG_LEVEL);
...
+	unsigned int flags = enable
+		? GPIO_INT_EDGE_TO_ACTIVE
+		: GPIO_INT_DISABLE;
 
-	gpio_pin_interrupt_configure(data->gpio,
-				     DT_INST_GPIO_PIN(0, int1_gpios),
-				     (enable
-				      ? GPIO_INT_EDGE_TO_ACTIVE
-				      : GPIO_INT_DISABLE));
+	gpio_pin_interrupt_configure_dt(&config->int1_gpio, flags);
```
Comment (gmarull) [inline drivers/sensor/bma280/bma280_trigger.c:29]: note: unrelated change, original code did not use an auxiliary variable

File: drivers/sensor/bma280/bma280.c [hunk 4]
```diff
@@ -161,8 +160,15 @@ int bma280_init(const struct device *dev)
 	return 0;
 }
 
-struct bma280_data bma280_driver;
+static struct bma280_data bma280_inst_data;
+
+static struct bma280_config bma280_inst_config = {
```
Comment (gmarull) [inline drivers/sensor/bma280/bma280.c:165]:
```suggestion
static const struct bma280_config bma280_inst_config = {
```

File: drivers/sensor/bma280/bma280.h [hunk 5]
```diff
@@ -142,6 +138,13 @@ struct bma280_data {
 #endif /* CONFIG_BMA280_TRIGGER */
 };
 
+struct bma280_config {
+	const struct i2c_dt_spec i2c;
+#ifdef CONFIG_BMA280_TRIGGER
+	const struct gpio_dt_spec int1_gpio;
```
Comment (gmarull) [inline drivers/sensor/bma280/bma280.h:144]:
```suggestion
	struct i2c_dt_spec i2c;
#ifdef CONFIG_BMA280_TRIGGER
	struct gpio_dt_spec int1_gpio;
```

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/sensor/bma280/bma280.c (hunk 2 above)
Comment (benjaminbjornsson) [reply drivers/sensor/bma280/bma280.c:168]: We certainly should, thanks!

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
